includes/shortcodes/class.llms.shortcodes.blocks.php

```diff
@@ -0,0 +1,231 @@
+<?php
+/**
+ * LifterLMS Shortcodes Blocks
+ *
+ * @package LifterLMS/Classes/Shortcodes
+ *
+ * @since [version]
+ * @version [version]
+ */
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * LLMS_Shortcodes_Blocks class.
+ *
+ * @since [version]
+ */
+class LLMS_Shortcodes_Blocks {
+
+	/**
+	 * Available shortcode blocks.
+	 *
+	 * @var array
+	 */
+	private $shortcodes = array(
+		'access-plan-button',
+		'checkout',
+		'courses',
+		'course-author',
+		'course-continue',
+		'course-meta-info',
+		'course-outline',
+		'course-prerequisites',
+		'course-reviews',
+		'course-syllabus',
+		'login',
+		'memberships',
+		'my-account',
+		'my-achievements',
+		'registration',
+	);
+
+	/**
+	 * Constructor.
+	 *
+	 * @since [version]
+	 *
+	 * @return void
+	 */
+	public function __construct() {
+		add_action( 'init', array( $this, 'register_blocks' ) );
+		add_action( 'after_setup_theme', array( $this, 'add_editor_styles' ) );
+		add_action( 'admin_enqueue_scripts', array( $this, 'enqueue_editor_styles' ) );
+		add_filter( 'llms_hide_registration_form', array( $this, 'show_form_preview' ) );
+		add_filter( 'llms_hide_login_form', array( $this, 'show_form_preview' ) );
+	}
+
+	/**
+	 * Registers shortcode blocks.
+	 *
+	 * @since [version]
+	 *
+	 * @return void
+	 */
+	public function register_blocks(): void {
+		foreach ( $this->shortcodes as $shortcode ) {
+			$block_dir = LLMS_PLUGIN_DIR . "blocks/$shortcode";
+
+			if ( file_exists( "$block_dir/block.json" ) ) {
+				register_block_type( $block_dir, [
+					'render_callback' => array( $this, 'render_block' ),
+				] );
+			}
+		}
+	}
+
+	/**
+	 * Loads front end CSS in the editor.
+	 *
+	 * @since [version]
+	 *
+	 * @return void
+	 */
+	public function add_editor_styles(): void {
+		add_editor_style( '../../plugins/lifterlms/assets/css/lifterlms.min.css' );
```

Comment:
can we use `LLMS_PLUGIN_DIR` here?

Reply:
I don't think we can use an absolute path here because `add_editor_style` paths are relative to the stylesheet directory. 

That being said, should we still build the path based on those variables in case it is not installed as `lifterlms` ?

Reply:
but if it's relative to the stylesheet directory, why don't we use:
`add_editor_style( './assets/css/lifterlms.min.css' );`
?

Reply:
That would work but only if Sky Pilot is active, otherwise we would still need to load the plugin styles. That also brings up a good point, we need to consider this in Sky Pilot as well

Reply:
I'm sorry I totally misread. You said stylesheet directory, I copied it and didn't think we're not a theme :D <3 (my past is hunting me ahah). Sorry, you're right, go ahead. <3